rundeckapp/grails-spa/packages/ui-trellis/src/app/components/job/resources/tests/NodeDetailSimple.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { mount } from "@vue/test-utils";
+import NodeDetailsSimple from "../NodeDetailsSimple.vue";
+
+jest.mock("@/library/rundeckService", () => ({
+  getRundeckContext: jest.fn().mockReturnValue({
+    rdBase: "mockRdBase",
+    projectName: "test-project",
+  }),
+  url: jest.fn().mockReturnValue({ href: "mockHref" }),
+}));
+const mockAttributes = {
+  osFamily: "Linux",
+  osName: "Ubuntu",
+  osVersion: "20.04",
+  osArch: "x86_64",
+  description: "Test Node 1",
+  "ui:status:text": "Healthy",
+  "ui:status:icon": "fa-check",
+  "ns1:attr1": "value1",
+  "ns1:attr2": "value2",
+  "ns1:attr3": "value3",
+  "ns1:attr4": "value4",
+  "namespace:key1": "value1",
+  "namespace:key2": "value2",
+  key3: "value3",
+  key4: "value4",
+  key5: "value5",
+  key6: "value6",
+  username: "user",
+  hostname: "host",
+};
+const mountNodeDetailsSimple = async (propsData = {}) => {
+  const wrapper = mount(NodeDetailsSimple, {
+    props: {
+      useNamespace: true,
+      nodeColumns: true,
+      attributes: mockAttributes,
+      tags: ["Tag1", "Tag2"],
+      ...propsData,
+    },
+    global: {
+      mocks: {
+        $t: (msg) => msg,
+      },
+    },
+  });
+  await wrapper.vm.$nextTick();
+  return wrapper;
+};
+describe("NodeDetailsSimple Component", () => {
+  it("renders node attributes", async () => {
+    const wrapper = await mountNodeDetailsSimple();
+    const attributes = wrapper.findAll("td.value, td.key, .setting");
+    const attributeTexts = attributes.map((attr) => attr.text());
+    const expectedAttributes = {
+      osFamily: "Linux",
+      osName: "Ubuntu",
+      osVersion: "20.04",
+      osArch: "x86_64",
+      description: "Test Node 1",
+      "ui:status:text": "Healthy",
+      "ui:status:icon": "fa-check",
+      "ns1:attr1": "value1",
+      "ns1:attr2": "value2",
+      "ns1:attr3": "value3",
+      "ns1:attr4": "value4",
+    };
+    Object.entries(expectedAttributes).forEach(([key, value]) => {
+      const attributeFound = attributeTexts.some((text) =>
+        text.includes(value),
+      );
+      expect(attributeFound).toBe(true);
+    });
+  });
```

Comment:
Few things:

1) the test where the props `useNamespace` and `nodeColumns` have a false value shouldn't be removed (if we get to the point of refactoring the component, then that case can be removed, but we should still cover it until then);

2) When these props are false, the majority of the attributes are rendered within the same tbody, and therefore picking it up with the `setting` class was acceptable, even though this approach wasn't asserting the attributes grouped under description/status/operating system/tags (because they do not use that class); But this isn't the case when the props are true. This test needs to be rewritten and ideally reorganized, asserting things within the groups they belong to instead of having all values in one single constant. Suggestion of structure:

- test selects all the general attributes (the attributes that live in the first <tbody>, description/status/operating system/tags) and asserts them against the expectedGeneralAttributes;

- test selects all the attributes that don't have a namespace and asserts them against the expectedAttributesWithNoNamespaces and also checks if the right node-filter-link was rendered (if it's the one with the glyphicon-plus or the glyphicon-minus)

- And when useNamespace is true, the test selects all attributes with namespaces, asserts that the namespace.values.length rendered matches the number of nested attributes and then finally asserts the nested attributes against the expectedNestedAttributes;

This way, if one day something is broken, it's easier to identify what is going wrong/missing. The way the test currently is written, the error message only hints that an attribute wasn't found.